### ISSUE TITLE
`azurerm_storage_account` - `allow_nested_items_to_be_public` resource documentation update

### ISSUE THREAD

**Issue by arpitcpanchal** [hashicorp/terraform-provider-azurerm#16034]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
```
Terraform v1.1.0
on windows_amd64
+ provider registry.terraform.io/hashicorp/azurerm v3.0.1
+ provider registry.terraform.io/hashicorp/random v3.1.2
```

### Affected Resource(s)
`azurerm_storage_account`

### Expected Behaviour
As per the https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide, 
The field `allow_blob_public_access` is renamed to `allow_nested_items_to_be_public` to resolve confusion about what this field does.

### Actual Behaviour
However, the resource documentation is still reflecting old field and needs to be updated. https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#allow_blob_public_access
<img width="903" alt="image" src="https://user-images.githubusercontent.com/84825339/160033357-c9f0e244-5559-4997-8da7-eacf90383c43.png">
<img width="522" alt="image" src="https://user-images.githubusercontent.com/84825339/160033391-6757efba-0b19-4929-ac98-0e18f5786c5a.png">

**Comment by katbyte** [comment]
Hey @arpitcpanchal, thanks for raising this with us! We unfortunately missed that documentation update and have patched it in #16025 which we'll release tomorrow afternoon along with a couple other minor bug fixes. As such I'm going to close this, apologies for any confusion this may have caused 😓

**Comment by arpitcpanchal** [comment]
Thanks @katbyte
